engine/src/main/resources/assets/shaders/lightGeometryPass_frag.glsl

```diff
@@ -89,8 +89,11 @@ void main() {
 
 #if defined (CLOUD_SHADOWS) && !defined (VOLUMETRIC_LIGHTING)
         // TODO: Add shader parameters for this...
-        float cloudOcclusion = clamp(texture2D(texSceneClouds, (worldPosition.xz + cameraPosition.xz) * 0.005 + timeToTick(time, 0.004)).r * 4,0,.4);
+        // Get the preconfigured value from the randomized texture, sampling a value from it to determine how much cloud shadow there will be.
+        // Clamp the value so that clouds do not turn the surface black.
+        float cloudOcclusion = clamp(texture2D(texSceneClouds, (worldPosition.xz + cameraPosition.xz) * 0.005 + timeToTick(time, 0.002)).r * 10,0.6,1);
```

Comment:
2 things here, the first is to slow down the clouds so they dont zip past quite as fast (the 0.002 value).  The second is to fix a bug where the non coud areas were being shadowed because we were clamping the opposite way.  You will notice a stark difference in the sand brightness in the "before" screenshots.

Before with clouds:
![terasology-160406170140-1280x720](https://cloud.githubusercontent.com/assets/5922109/14394521/819ac418-fd81-11e5-82b2-f00bc2996f03.jpg)

Before without clouds:
![terasology-160406170158-1280x720](https://cloud.githubusercontent.com/assets/5922109/14394523/85413dc2-fd81-11e5-9596-b4d5cad2b817.jpg)

After with clouds:
![terasology-160408115515-1280x720](https://cloud.githubusercontent.com/assets/5922109/14394542/a4beff22-fd81-11e5-8809-813971342018.jpg)

After without clouds:
![terasology-160408115448-1280x720](https://cloud.githubusercontent.com/assets/5922109/14394551/a89b4f9c-fd81-11e5-9d16-50f57d980635.jpg)